tests/index.js

```diff
@@ -11,38 +11,51 @@ mongoose.connect(config.mongo.data.uri, {useMongoClient: true});
 const awaitLastBlock = require('./helpers/awaitLastBlock'),
   net = require('net'),
   WebSocket = require('ws'),
-  path = require('path'),
   Web3 = require('web3'),
   web3 = new Web3(),
   expect = require('chai').expect,
   accountModel = require('../models/accountModel'),
+  blockModel = require('../models/blockModel'),
   amqp = require('amqplib'),
   Stomp = require('webstomp-client'),
   ctx = {};
 
+let accounts;
+
+
+
+
 describe('core/block processor', function () {
 
+  const clearMongoBlocks = async function () {
+      await accountModel.remove()
+      await blockModel.find({hash: {$exists: false}}).remove()
+  };
+
   before(async () => {
+    clearMongoBlocks();
+    
     let provider = new Web3.providers.IpcProvider(config.web3.uri, net);
     web3.setProvider(provider);
-
+    accounts = await Promise.promisify(web3.eth.getAccounts)();
+    
     return await awaitLastBlock(web3);
   });
 
   after(() => {
+    clearMongoBlocks();
```

Comment:
forgot await keyword. In case, mongo close the connection earlier - you won't be able to cleanup anything!